test/e2e/workload_test.go

```diff
@@ -616,3 +654,224 @@ var _ = Describe("Statefulset", func() {
 	})
 
 })
+
+var _ = Describe("DaemonSet", func() {
+	BeforeEach(func() {
+		f = root.Invoke()
+	})
+	JustBeforeEach(func() {
+		cred = f.SecretForLocalBackend()
+		if missing, _ := BeZero().Match(cred); missing {
+			Skip("Missing repository credential")
+		}
+		pvc = f.GetPersistentVolumeClaim()
+		err = f.CreatePersistentVolumeClaim(pvc)
+		Expect(err).NotTo(HaveOccurred())
+		repo = f.Repository(cred.Name, pvc.Name)
+
+		backupCfg = f.BackupConfiguration(repo.Name, targetref)
+		rules = []v1beta1.Rule{
+			{
+				Paths: []string{
+					framework.TestSourceDataMountPath,
+				},
+			},
+		}
+		restoreSession = f.RestoreSession(repo.Name, targetref, rules)
+	})
+	AfterEach(func() {
+		err = f.DeleteSecret(cred.ObjectMeta)
+		Expect(err).NotTo(HaveOccurred())
+		err = framework.WaitUntilSecretDeleted(f.KubeClient, cred.ObjectMeta)
+		Expect(err).NotTo(HaveOccurred())
+	})
+	var (
+		testDaemonSetBackup = func() {
+			By("Create DaemonSet" + daemonset.Name)
+			_, err := f.CreateDaemonSet(daemonset)
+			Expect(err).NotTo(HaveOccurred())
+			err = f.WaitUntilDaemonPodReady(daemonset.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+			f.EventuallyPodAccessible(daemonset.ObjectMeta).Should(BeTrue())
+
+			By("Creating Sample data inside pod")
+			err = f.CreateSampleDataInsideWorkload(daemonset.ObjectMeta, apis.KindDaemonSet)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Reading sample data from /source/data mountPath inside workload")
+			sampleData, err = f.ReadSampleDataFromFromWorkload(daemonset.ObjectMeta, apis.KindDaemonSet)
```

Comment:
How do you know sample data has been created successfully? Check if `sampleData` is not empty.